src/cmds/openapi/refs.ts

```diff
@@ -0,0 +1,235 @@
+      }
+
+      const schemaName = refParts[3];
+      const propertyName = refParts[5];
+      const schema = schemas[schemaName];
+      const property = schema?.properties?.[propertyName];
+
+      if (!schema || !property) {
+        throw new Error(`Schema or property not found for path: ${refPath}`);
+      }
+
+      // Handle references within items in an array
+      let refSchemaName: string | undefined;
+      if (
+        refParts.length > 6 &&
+        refParts[6] === 'items' &&
+        property.type === 'array' &&
+        property.items &&
+        typeof property.items === 'object'
+      ) {
+        const itemsRefSchemaName = (property.items as IJsonSchema).$ref?.split('/')[3];
+        if (itemsRefSchemaName) {
+          refSchemaName = `${itemsRefSchemaName}Ref`;
+          property.items = { $ref: `#/components/schemas/${refSchemaName}` } as IJsonSchema;
+          createRefSchema(itemsRefSchemaName, refSchemaName);
+        }
+      } else {
+        // Handle direct reference
+        refSchemaName = property.$ref?.split('/')[3];
+        if (refSchemaName) {
+          const newRefSchemaName = `${refSchemaName}Ref`;
+          replaceRef(schemaName, propertyName, newRefSchemaName);
+          createRefSchema(refSchemaName, newRefSchemaName);
+        }
+      }
+    });
+  }
+
+  /**
+   * The main execution method for the command.
+   * @param {ZeroAuthCommandOptions<Options>} opts - Command options.
+   * @returns {Promise<string>} Result message.
+   */
+  async run(opts: ZeroAuthCommandOptions<Options>): Promise<string> {
+    await super.run(opts);
+    const { spec } = opts;
+    if (!spec) {
+      return 'File path is required.';
+    }
+
+    const openApiData = OpenAPISolvingCircularityAndRecursiveness.readOpenApiFile(spec);
+    const circularRefs = await OpenAPISolvingCircularityAndRecursiveness.getCircularRefsFromOas(openApiData);
+
+    if (circularRefs.length === 0) {
+      return 'The file does not contain circular or recursive references.';
+    }
+
+    if (openApiData.components?.schemas && circularRefs.length > 0) {
+      OpenAPISolvingCircularityAndRecursiveness.replaceCircularRefs(openApiData.components.schemas, circularRefs);
+
+      let remainingCircularRefs = await OpenAPISolvingCircularityAndRecursiveness.getCircularRefsFromOas(openApiData);
+      let iterationCount = 0;
+
+      while (remainingCircularRefs.length > 0 && iterationCount < 5) {
+        OpenAPISolvingCircularityAndRecursiveness.replaceCircularRefs(
+          openApiData.components.schemas,
+          remainingCircularRefs,
+        );
+        remainingCircularRefs = remainingCircularRefs.length > 0 ? [] : remainingCircularRefs;
+        iterationCount += 1;
+      }
+
+      if (iterationCount >= 5) {
+        return 'Maximum iteration limit reached. Some circular references may remain unresolved.';
+      }
+    }
+
+    OpenAPISolvingCircularityAndRecursiveness.writeOpenApiFile(spec, openApiData);
```

Comment:
rather than overwriting the existing file, can you add a prompt (and a corresponding `--outputPath` flag) so users can specify what the updated file path should be? see here for an example:

https://github.com/readmeio/rdme/blob/5db25a1b5bec9c26966eda5cd1d616a4de668601/src/cmds/openapi/convert.ts#L66-L85